appender.go

```diff
@@ -187,17 +188,52 @@ func (a *Appender) Flush() error {
 		return getError(errAppenderFlush, invalidatedAppenderError(err))
 	}
 
-	if mapping.AppenderFlush(a.appender) == mapping.StateError {
-		err := getDuckDBError(mapping.AppenderError(a.appender))
+	if err := a.flush(); err != nil {
 		return getError(errAppenderFlush, invalidatedAppenderError(err))
 	}
 
 	return nil
 }
 
+// FlushWithCancel flushes the data chunks to the underlying table and clears the internal cache.
+// Does not close the appender, even if it returns an error. Unless you have a good reason to call this,
+// call CloseWithCancel when you are done with the appender.
+// Takes a context for cancellation.
+func (a *Appender) FlushWithCancel(ctx context.Context) error {
+	if err := a.appendDataChunk(); err != nil {
+		return getError(errAppenderFlush, invalidatedAppenderError(err))
+	}
+
+	if err := a.flushWithCancel(ctx); err != nil {
+		return getError(errAppenderFlush, invalidatedAppenderError(err))
+	}
+
+	return nil
+}
+
+func (a *Appender) Clear() error {
```

Comment:
I think we also need to clear the latest data chunk in the appender? Otherwise, we'll still try to flush that one, no?

```Go
	// The chunk to append to.
	chunk DataChunk
```